Python/optimizer_bytecodes.c

```diff
@@ -886,6 +886,44 @@ dummy_func(void) {
         }
     }
 
+    op(_CALL_ISINSTANCE, (callable, self_or_null, args[oparg] -- res)) {
+        if (sym_is_null(self_or_null) || sym_is_not_null(self_or_null)) {
+            if (sym_is_not_null(self_or_null)) {
+                args--;
+            }
+            JitOptSymbol *cls_sym = args[1];
+            JitOptSymbol *inst_sym = args[0];
```

Comment:
Careful, we need to check the argument count before doing any of this. Something like `if (oparg + sym_is_not_null(self_or_null) == 2) { ... }`.

That's actually an argument in favor of reworking this instruction to only specialize for `self_or_null == NULL` and `oparg == 2`, since it makes this a bit less painful. Your call what order you want to do it in.

Reply:
I opened https://github.com/python/cpython/pull/133339 which restricts the specialization. With that, we'll be able to get rid of the fiddly index adjustments :)